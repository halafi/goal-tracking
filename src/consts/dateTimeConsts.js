export const GOAL_DATE_TIME = 'YYYY-MM-DDTHH:mm'
