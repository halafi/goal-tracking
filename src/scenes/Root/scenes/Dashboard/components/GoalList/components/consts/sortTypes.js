export const GOAL_SORT_TYPES = [
  {
    value: 'default',
    label: 'Default',
  },
  {
    value: 'name',
    label: 'Name',
  },
  {
    value: 'progress',
    label: 'Progress',
  },
]

export type SortType = 'default' | 'name' | 'progress'
