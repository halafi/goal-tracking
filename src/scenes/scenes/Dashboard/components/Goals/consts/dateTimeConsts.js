export const GOAL_DATE_TIME = 'YYYY-MM-DD'
