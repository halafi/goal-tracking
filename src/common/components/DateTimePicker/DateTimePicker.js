// @flow

import React from 'react'
import moment from 'moment/moment'
import TextField from 'material-ui/TextField'
import { GOAL_DATE_TIME } from '../../consts/dateTimeConsts'

type Props = {
  id: string,
  className: ?string,
  disabled: boolean,
  label: ?string,
  onChange: any => void,
  value: string,
}

const DateTimePicker = ({ id, label, value, onChange, className, disabled }: Props) => (
  <TextField
    id={id}
    label={label}
    type="datetime-local"
    value={moment(value).format(GOAL_DATE_TIME)}
    InputLabelProps={{
      shrink: true,
    }}
    onChange={onChange}
    className={className}
    disabled={disabled}
  />
)

export default DateTimePicker
