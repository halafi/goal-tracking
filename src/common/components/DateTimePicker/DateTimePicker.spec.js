import React from 'react'
import { shallow } from 'enzyme'
import moment from 'moment'

import DateTimePicker from './DateTimePicker'
import { GOAL_DATE_TIME } from '../../consts/dateTimeConsts'

describe('DateTimePicker', () => {
  const dateNow = Date.now
  beforeEach(() => {
    Date.now = jest.fn(() => 1518110488752) // lock at "2018-02-08T18:21:45+01:00"
  })
  afterEach(() => {
    Date.now = dateNow
  })
  it('renders component', () => {
    const wrapper = shallow(
      <DateTimePicker
        id="test-picker"
        label="Date Time"
        value={moment()
          .subtract(5, 'd')
          .valueOf()}
        onChange={null}
      />,
    )
    expect(wrapper.getElement()).toMatchSnapshot()
    expect(wrapper.props()).toMatchSnapshot()
  })
  it('value change work', () => {
    const val1 = moment()
      .subtract(5, 'd')
      .valueOf()

    const val2 = moment()
      .add(5, 'd')
      .valueOf()

    let wrapper = shallow(
      <DateTimePicker id="test-picker" label="Date Time" value={val1} onChange={null} />,
    )
    expect(moment(wrapper.props().value).format(GOAL_DATE_TIME)).toEqual('2018-02-03T18:21')
    wrapper = shallow(
      <DateTimePicker id="test-picker" label="Date Time" value={val2} onChange={null} />,
    )
    expect(moment(wrapper.props().value).format(GOAL_DATE_TIME)).toEqual('2018-02-13T18:21')
  })
})
