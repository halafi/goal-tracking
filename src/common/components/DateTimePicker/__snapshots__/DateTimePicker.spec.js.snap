// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker renders component 1`] = `
<TextField
  InputLabelProps={
    Object {
      "shrink": true,
    }
  }
  className=""
  disabled={false}
  id="test-picker"
  label="Date Time"
  onChange={null}
  required={false}
  select={false}
  type="datetime-local"
  value="2018-02-03T18:21"
/>
`;

exports[`DateTimePicker renders component 2`] = `
Object {
  "InputLabelProps": Object {
    "shrink": true,
  },
  "className": "",
  "disabled": false,
  "id": "test-picker",
  "label": "Date Time",
  "onChange": null,
  "required": false,
  "select": false,
  "type": "datetime-local",
  "value": "2018-02-03T18:21",
}
`;
